modules/dhcp_common/subnet.rb

```diff
@@ -62,28 +64,18 @@ def netmask_to_i
       @netmask_to_i ||= Proxy::DHCP.ipv4_to_i(netmask)
     end
 
-    def get_index_and_lock filename
+    def get_index filename
       # Store for use in the unlock method
       @filename = "#{Dir::tmpdir}/#{filename}"
-      @lockfile = "#{@filename}.lock"
-
-      # Loop if the file is locked
-      Timeout::timeout(30) { sleep 0.1 while File.exist? @lockfile }
-
-      # Touch the lock the file
-      File.open(@lockfile, "w") {}
-
       @file = File.new(@filename,'r+') rescue File.new(@filename,'w+')
-
       # this returns the index in the file
       return @file.readlines.first.to_i rescue 0
```

Comment:
I think this whole thing should be re-written with an in-memory implementation. Additionally, there are issues with re-using the index: the list of free ips changes as hosts get added/deleted. A position in such a list isn't a great index.